File: be2-scala/src/main/scala/ch/epfl/pop/model/network/method/Publish.scala
package ch.epfl.pop.model.network.method

import ch.epfl.pop.json.HighLevelProtocol._
import ch.epfl.pop.model.network.Parsable
import ch.epfl.pop.model.network.method.message.Message
import ch.epfl.pop.model.objects.Channel
import spray.json._

final case class Publish(override val channel: Channel, override val message: Message) extends ParamsWithMessage(channel, message)

object Publish extends Parsable {
  def apply(channel: Channel, message: Message): Publish = {
    new Publish(channel, message)
  }

  override def buildFromJson(payload: String): Publish = payload.parseJson.asJsObject.convertTo[Publish]
}

File: be1-go/message/query/method/publish.go
package method

import (
	"popstellar/message/query"
	"popstellar/message/query/method/message"
)

// Publish defines a JSON RPC publish message
type Publish struct {
	query.Base

	ID int `json:"id"`

	Params PublishParams `json:"params"`
}

type PublishParams struct {
	Channel string          `json:"channel"`
	Message message.Message `json:"message"`
}
